daemon/emer-aggregate-tally.c

```diff
@@ -0,0 +1,373 @@
+                                          G_FILE_QUERY_INFO_NONE,
+                                          NULL,
+                                          &error);
+
+  if (error)
+    {
+      if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+        g_warning ("Error listing tallies from %s: %s",
+                   tallies_path, error->message);
+      return;
+    }
+
+  while (TRUE)
+    {
+      g_autoptr(GVariant) variant = NULL;
+      g_autoptr(GVariant) cache_key = NULL;
+      g_autoptr(GVariant) event_id = NULL;
+      g_autoptr(GVariant) aggregate_key = NULL;
+      g_autoptr(GVariant) payload = NULL;
+      g_autoptr(GError) local_error = NULL;
+      g_autofree gchar *contents = NULL;
+      g_autofree gchar *path = NULL;
+      EmerTallyIterResult result;
+      GFileInfo *info;
+      GFile *file;
+      guint32 counter;
+      guint32 unix_user_id;
+      size_t size;
+
+      g_file_enumerator_iterate (enumerator, &info, &file, NULL, &local_error);
+
+      if (local_error)
+        {
+          if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+            g_warning ("Error listing tallies from %s: %s",
+                       tallies_path, local_error->message);
+          return;
+        }
+
+      if (!file || !info)
+        break;
+
+      if (g_file_info_get_file_type (info) != G_FILE_TYPE_REGULAR)
+        continue;
+
+      path = g_file_get_path (file);
+      g_file_get_contents (path, &contents, &size, &local_error);
+
+      if (local_error)
+        {
+          g_warning ("Error loading tally from file %s: %s",
+                     path, local_error->message);
+          continue;
+        }
+
+      variant = g_variant_new_from_data (G_VARIANT_TYPE ("(vvuumv)"),
+                                         contents, size,
+                                         FALSE,
+                                         NULL, NULL);
+      if (local_error)
+        {
+          g_autoptr(GError) deletion_error = NULL;
+
+          g_warning ("Error parsing tally from file %s: %s",
+                     path, local_error->message);
+
+          g_file_delete (file, NULL, &deletion_error);
+          if (deletion_error)
+            g_warning ("Error deleting corrupted tally file %s: %s",
+                       path, deletion_error->message);
+          continue;
+        }
+
+      g_variant_get (variant,
+                     "(vvuumv)",
+                     &event_id,
+                     &aggregate_key,
+                     &unix_user_id,
+                     &counter,
+                     &payload);
+
+      result = func (unix_user_id, event_id,
+                     aggregate_key, payload,
+                     counter, date, user_data);
+
+      if (result & EMER_TALLY_ITER_DELETE)
+        {
+          g_autoptr(GError) deletion_error = NULL;
+
+          g_warning ("Error parsing tally from file %s: %s",
+                     path, local_error->message);
```

Comment:
`local_error` may be NULL here. I think this block is copy-pasta from higher up.